lib/capybara/poltergeist/client.rb

```diff
@@ -102,14 +103,20 @@ def command
     # lose all the output from child. Process.popen can be used here and seems
     # it works with JRuby but I've experienced strange mistakes on Rubinius.
     def redirect_stdout
-      prev = STDOUT.dup
-      $stdout = @write_io
-      STDOUT.reopen(@write_io)
-      yield
-    ensure
-      STDOUT.reopen(prev)
-      $stdout = STDOUT
-      prev.close
+      if Capybara::Poltergeist.jruby?
```

Comment:
Good question. If not, we could make this test more strict.

Separately, the thread safety issue still exists for JRuby (and Rubinius, if it doesn't support the `:out` option). I think it's outside the scope of this PR, but it may be worth looking into an alternate solution there, e.g. `fork`, then redirect STDOUT, then `exec` (assuming those VMs support `fork` and `exec`) rather than `Process.spawn`.